Decode Remote Google Logs

KevYuen opened the pull request:
The `Blob.download_as_string` function returns a byte which cause
the log result to be displayed in a single line like:

b"line1\nline2"

instead of

line1
line2

added an isinstance check to make sure it doesn't break if it
returns string in some case and not others

all the logs is in a single line: 

<img width="959" alt="image" src="https://user-images.githubusercontent.com/4744103/102393049-daec3900-3fa5-11eb-8b37-361adac86ba4.png">

Version: 2.0.0rc3

TODO: 
---
- [x] Update test to include case with byte return value instead of string (apologies, fighting with breeze)

KevYuen committed:
decode remote google logs before returning

The `Blob.download_as_string` function returns a byte which cause
the log result to be displayed in a single line like:

b"line1\nline2"

instead of

line1
line2

added an isinstance check to make sure it doesn't break if it
returns string in some case and not others

KevYuen committed:
add google provide remote log test case with byte

parameterize the remote log task to test both case where
byte and string is returned from download_as_string

KevYuen committed:
fix import order

Put the parameterized import with builtin lib import by accident

turbaszek commented on airflow/providers/google/cloud/log/gcs_task_handler.py:
```diff
@@ -153,6 +153,10 @@ def _read(self, ti, try_number, metadata=None):
             remote_log = blob.download_as_string()
+
+            if isinstance(remote_log, bytes):
+                remote_log = remote_log.decode()
```
I would say that this should be part of `download_as_string` method. The name suggest that the output should be a `str` not `bytes` so fixing it there may solve some issues in future.

KevYuen replied:
You're right, looks like the method is deprecated and it is now an alias for `download_as_byte`

https://googleapis.dev/python/storage/latest/blobs.html#google.cloud.storage.blob.Blob.download_as_string

Will update the PR

KevYuen committed:
replace download_as_string with download_as_byte

The function download_as_string is deprecated in google-cloud-storage and
is an alias for download_as_byte. Use download_as_byte directly

KevYuen committed:
change test download_as_byte return value to byte

KevYuen committed:
fix write to remote on close test

The write to remote should no longer fail if the log download is byte

KevYuen committed:
Merge remote-tracking branch 'origin' into gcp-provider-decode-logs

KevYuen committed:
correct function download_as_bytes typo

the function download_as_byte should be download_as_bytes

KevYuen committed:
trigger GitHub actions

KevYuen committed:
Merge remote-tracking branch 'origin' into gcp-provider-decode-logs
